src/parcsr_ls/par_ilu_setup.c

```diff
@@ -7,6 +7,34 @@
 #include "_hypre_parcsr_ls.h"
 #include "par_ilu.h"
 
+#include "_hypre_utilities.hpp"
```

Comment:
@oseikuffuor1 @liruipeng This seems like correct placement to me, but not the `par_ilu.h` include before it.

Reply:
@rfalgout I agree

Reply:
@falgout, @ruipeng, I'm not sure I understand this. Is the suggestion to *not* include par_ilu.h here?

Reply:
What I think I meant here is basically the same as some of the other comments.  Should 'par_ilu.h' be in '_hypre_parcsr_ls.h' instead?  We try to not include too many header files in hypre source files, and instead isolate the complexities of dealing with header files to the '_hypre_*.h' header files.

Reply:
@rfalgout Ok that makes sense. I can see about moving it there. I think there are a few other instances like this in parcsr_ls (for example in par_amg_setup.c, we include par_amg.h), but I assume the goal is the clean this up later. Thanks.

Reply:
@rfalgout The reason we do not put `par_ilu.h` inside `_hypre_parcsr_ls.h` is that `par_ilu.h` has cusparse stuff and we don't want cusparse is seen by all the files that include `_hypre_parcsr_ls.h`.

Reply:
Okay, then should we create a `_hypre_parcsr_ls.hpp` file instead?  Didn't we start doing this elsewhere in hypre?  Note that this is not critical to finishing this pull request.  We can revisit it later.  I don't want to slow this down.

Reply:
@rfalgout Yes, we have `_hypre_struct_mv.hpp` and `../utilities/_hypre_utilities.hpp`. It makes sense to do the same thing here. It is just needed by ILU now. We may leave it as is now and revisit it later. Thanks!